core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -125,6 +125,13 @@ public void setTermsLookupFetchService(@Nullable  TermsLookupFetchService termsL
         this.termsLookupFetchService = termsLookupFetchService;
     }
 
+    /**
+     * For testing purposes only - ensures that e.g. for GeoBoundingBoxQueries we can inject a mocked service.
+     * */
+    void setFieldDataService(IndexFieldDataService fieldDataService) {
+        this.fieldDataService = fieldDataService;
+    }
```

Comment:
Just re-iterating to make sure I understand you correctly: Essentially moving back to what we had here:

https://github.com/MaineC/elasticsearch/commit/7fca091f97f0941b9ff5bab13833cab3efcfbed6#diff-a38e311dff061a2d8c3888dc0d1eb39dR174 except for not using a mock framework but an explicitly mocked class?